### PR TITLE
primes Carousel.js to work w/ Lists + fixes existing bugs

### PR DESCRIPTION
**Pull request by mekarpeles**
Upgrades our Carousel.js loadmore functionality to be more robust to values returned by API.

### Technical

This fixes cases where:

1. our availability API returns `status: private` e.g. https://archive.org/services/availability?identifier=Hello.
2. prevents breakage when `book` doesn't have an `identifier` / `ocaid`
3. improves availability accuracy by looking for additional fields for availability options

### Testing

Existing carousels should work without much change. @seabelis may have noticed some carousels which **don't** have `ocaid`s (e.g. Not In Library) likely break existing carousels when we try to Load More pages via ajax. This should be solved.

This PR is also a pre-requisite for passing list keys to `QueryCarousels`

### Stakeholders

@cdrini @jamesachamp

### CODE REVIEW

**Review by mekarpeles — COMMENTED** [review]
old feedback, disregard

File: openlibrary/plugins/openlibrary/js/carousel/Carousel.js [hunk 1]
```diff
@@ -160,11 +174,8 @@ const Carousel = {
                     $.ajax({
                         url: url,
                         type: 'GET',
-                        success: function(subject_results) {
-                            var works = subject_results.works;
-                            if (!works) {
-                                works = subject_results.docs;
-                            }
+                        success: function(results) {
+                            var works = results.works || results.docs;
```
Comment (jamesachamp) [inline openlibrary/plugins/openlibrary/js/carousel/Carousel.js:178]:
```suggestion
                            const works = results.works || results.docs;
```
We should move away from using `var` to declare variables.  `const` is good in this case as `works` will not be reassigned.

File: openlibrary/plugins/openlibrary/js/carousel/Carousel.js [hunk 2]
```diff
@@ -77,22 +77,36 @@ const Carousel = {
...
-            var url = (cls == 'cta-btn--available') ?
+            let availabilityStatus = availabilityStatuses[availability.status];
+            let cls = availabilityStatus.cls;
+            let cta = availabilityStatus.cta;
+            let url = (cls == 'cta-btn--available') ?
                 (`/borrow/ia/${ocaid}`) : (cls == 'cta-btn--unavailable') ?
-                    (`/books/${work.availability.openlibrary_edition}`) : work.key;
-            var cta = availabilityStatuses[availability].cta;
-            var isClickable = availability == 'error' ? 'disabled' : '';
+                    (`/books/${openlibrary_edition}`) : work.key;
```
Comment (jamesachamp) [inline openlibrary/plugins/openlibrary/js/carousel/Carousel.js:108]: Are the URLs for preview buttons handled here?  It looks like they'll redirect to `/borrow/ia/${ocaid}`.

**Review by jamesachamp — CHANGES_REQUESTED** [review]
Aside from the `var` and the question about preview button URLs, this looks good to me.

File: openlibrary/plugins/openlibrary/js/carousel/Carousel.js (hunk 2 above)
Comment (jamesachamp) [reply openlibrary/plugins/openlibrary/js/carousel/Carousel.js:108]: Disregard this.  I didn't realize that "Preview" was merely the call to action for private books, and not an actual preview button.
